src/DynamoCore/Models/AnnotationModel.cs

```diff
@@ -120,7 +123,7 @@ public Double TextBlockHeight
             {
                 textBlockHeight = value;                
                 Y = InitialTop - textBlockHeight;
-                Height = InitialHeight + textBlockHeight;
+                Height = InitialHeight + textBlockHeight - minHeight;
             }
```

Comment:
I have this TextBlockHeight on Model layer, because the boundary is calculated on the Model Layer and this is so much used only on the model layer.